Nos/Views/Components/Images/ImageViewer.swift

```diff
@@ -0,0 +1,206 @@
+import SDWebImageSwiftUI
+import SwiftUI
+
+/// A viewer for images. Supports full-screen zoom and panning.
+struct ImageViewer: View {
+    /// The URL of the image to display.
+    let url: URL
+
+    @Environment(\.dismiss) private var dismiss
+    
+    /// The current zoom scale of the image.
+    @State private var scale: CGFloat = 1.0
+
+    /// The offset of the image. This is updated when the user has zoomed and is panning up, down, left, and right.
+    @State private var offset: CGSize = .zero
+
+    /// The previous offset of the image.
+    /// - SeeAlso: `offset`
+    @State private var lastOffset: CGSize = .zero
+
+    /// The size of the image. Will be set to a non-zero value when the image has loaded.
+    @State private var imageSize: CGSize = .zero
+    
+    /// The maximum zoom scale for the image.
+    private let maxScale: CGFloat = 10.0
+    
+    /// The minimum zoom scale for the image.
+    private let minScale: CGFloat = 1.0
+
+    var body: some View {
+        ZStack {
+            Color.imageBackground
+                .ignoresSafeArea()
+
+            GeometryReader { geometry in
+                WebImage(url: url)
+                    .onSuccess { image, _, _ in
+                        imageSize = image.size
+                    }
+                    .resizable()
+                    .aspectRatio(contentMode: .fit)
+                    .scaleEffect(scale)
+                    .offset(x: offset.width, y: offset.height)
+                    .gesture(
+                        DragGesture()
+                            .onChanged { value in
+                                withAnimation {
+                                    offset = CGSize(
+                                        width: lastOffset.width + value.translation.width * 1.5,
+                                        height: lastOffset.height + value.translation.height * 1.5
```

Comment:
@martindsq you said this feels weird, like the image is moving too much. We can adjust these values and see how it works for you:

```suggestion
                                        width: lastOffset.width + value.translation.width,
                                        height: lastOffset.height + value.translation.height
```